packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1005,16 +1110,22 @@ class ContainerLayer extends Layer {
 
   @override
   void detach() {
+    assert(!_debugMutationsLocked);
     super.detach();
     Layer? child = firstChild;
     while (child != null) {
       child.detach();
       child = child.nextSibling;
     }
+    // Detach indicates that we may never be composited again. Clients
+    // interested in observing composition will need to get updated to show
+    // Children fired them already in child.detach().
```

Comment:
something is odd here. Is the sentence on the previous line incomplete?

Reply:
Yes, fixed that.